decentraland/screens/MyAssetsScreen.js

```diff
@@ -1,25 +1,36 @@
 import React from "react";
 import { connect } from "react-redux";
-import { setMyAssetsList } from "../redux/actions";
+import { removeFromMyAssetsList, appendToMyAssetsList } from "../redux/actions";
 import PropTypes from "prop-types";
 import MyAssetsList from "@presentational/MyAssetsList";
 import MyAssetsListItem from "@presentational/MyAssetsListItem";
 import { listsAreEqual, getContracts } from "@helpers";
 import { generateAssetFromId } from "@helpers/decentraland";
 import DecentralandUtils from "tasit-sdk/dist/helpers/DecentralandUtils";
+import MyAssetStatus from "@constants/MyAssetStatus";
+const { BOUGHT } = MyAssetStatus;
 
 export class MyAssetsScreen extends React.Component {
   componentDidMount = async () => {
-    const { account, myAssets: assetsFromState, setMyAssetsList } = this.props;
+    const {
+      account,
+      myAssets: assetsFromState,
+      removeFromMyAssetsList,
+      appendToMyAssetsList,
+    } = this.props;
     if (account) {
       const { address } = account;
+
+      // Note: If an stored asset hasn't a status, assuming 'bought'.
+      const boughtAssets = assetsFromState
+        .map(asset => (!asset.status ? { ...asset, status: BOUGHT } : asset))
+        .filter(asset => asset.status === BOUGHT);
+
       const assetsFromBlockchain = await this._getAssetsFromBlockchain(address);
-      const shouldUpdate = !listsAreEqual(
-        assetsFromBlockchain,
-        assetsFromState
-      );
+      const shouldUpdate = !listsAreEqual(assetsFromBlockchain, boughtAssets);
       if (shouldUpdate) {
-        setMyAssetsList(assetsFromBlockchain);
+        removeFromMyAssetsList(boughtAssets);
+        appendToMyAssetsList(assetsFromBlockchain);
```

Comment:
If an asset was locally there but ends up being removed as a result of this, I think we should log something at a minimum, and eventually add some UI indication that something unexpected happened

Reply:
Since it's related to `Toast` make sense extract that?

Reply:
Hmm, I think that may sound good but I'm not totally sure what you mean.

Reply:
> add some UI indication that something unexpected happened

I thought to use `Toast` component for that, but it isn't working properly right now.

Reply:
Oh yeah. Plus I think we're maybe using Toasts a little bit too much? They work well as a placeholder for future UI elements, though. Maybe just log something for now and add a TODO comment in the code to reflect it in the UI?

Reply:
OK

Reply:
Done. Should we log more detailed info (e.g.: which assetIds aren't on the blockchain)?

Reply:
Sure!

Reply:
Done.

Reply:
Thanks! Resolving this